src/requirements/data/majors/foodsci.ts

```diff
@@ -0,0 +1,276 @@
+    source: 'https://cals.cornell.edu/education/degrees-programs/food-science-major-minor',
+    checker: includesWithSingleRequirement('BIOMG 3310', 'BIOMG 3350', 'NS 3200'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Physics',
+    description: 'PHYS 1101 or PHYS 2207',
+    source: 'https://cals.cornell.edu/education/degrees-programs/food-science-major-minor',
+    checker: includesWithSingleRequirement('PHYS 1101', 'PHYS 2207'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Core Courses',
+    description:
+      'These are the core courses required for every concentration. ' +
+      'FDSC 1101, FDSC 1102, NS 1150, FDSC 2000, FDSC 2100, FDSC 2110, ' +
+      'FDSC 3940, FDSC 3960, FDSC 4000, FDSC 4170,FDSC 4210',
+    source: 'https://cals.cornell.edu/education/degrees-programs/food-science-major-minor',
+    checker: includesWithSubRequirements(
+      ['FDSC 1101'],
+      ['FDSC 1102'],
+      ['NS 1150'],
+      ['FDSC 2000'],
+      ['FDSC 2100'],
+      ['FDSC 2110'],
+      ['FDSC 3940'],
+      ['FDSC 3960'],
+      ['FDSC 4000'],
+      ['FDSC 4170'],
+      ['FDSC 4210']
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+    slotNames: [
+      'FDSC 1101',
+      'FDSC 1102',
+      'NS 1150',
+      'FDSC 2000',
+      'FDSC 2100',
+      'FDSC 2110',
+      'FDSC 3940',
+      'FDSC 3960',
+      'FDSC 4000',
+      'FDSC 4170',
+      'FDSC 4210',
+    ],
+  },
+  {
+    name: 'Concentration',
+    description:
+      'Students are required to complete one concentration from Science, Safety and Business.\nWarning: For the optional electives in each concentration, we do not check whether the courses selected fulfil the minimum credits needed for that category.',
+    source: 'https://cals.cornell.edu/education/degrees-programs/food-science-major-minor',
+    fulfilledBy: 'toggleable',
+    // TODO: Can change Science and Business options to be fulfilled by credits instead, however would lose slot names.
+    fulfillmentOptions: {
+      Science: {
+        description:
+          'Required FDSC courses, and 3 >4000 FDSC electives, with at least 3 credits from group 1 and 2 credits from group 2.',
+        counting: 'courses',
+        checker: includesWithSubRequirements(
+          ['FDSC 3950'],
+          ['FDSC 4100'],
+          ['FDSC 4190'],
+          ['FDSC 4230'],
+          ['FDSC 4010', 'FDSC 4040', 'FDSC 4050', 'FDSC 4220', 'FDSC 4250', 'FDSC 4400'],
+          [
+            'FDSC 4020',
+            'FDSC 4110',
+            'FDSC 4500',
+            'FDSC 4510',
+            'FDSC 4880',
+            'FDSC 5970',
+            'FDSC 6950',
+            'FDSC 4010',
+            'FDSC 4040',
+            'FDSC 4050',
+            'FDSC 4220',
+            'FDSC 4250',
+            'FDSC 4400',
+          ]
```

Comment:
If the user needs 3 >4000 FDSC electives, would we need another subrequirement slot that is "Group 1 or Group 2 >4000 electives" where it includes courses in both of these arrays?

Reply:
Right, that makes sense. Done!